go/multipooler/manager/rpc_consensus.go

```diff
@@ -0,0 +1,284 @@
+	// Update our term and reset vote
+	newTerm := &pgctldpb.ConsensusTerm{
+		CurrentTerm: req.Term,
+		VotedFor:    nil,
+	}
+	if err := SetTerm(pm.config.PoolerDir, newTerm); err != nil {
+		return nil, fmt.Errorf("failed to update term: %w", err)
+	}
+	pm.mu.Lock()
+	pm.consensusTerm = newTerm
+	pm.mu.Unlock()
+
+	currentTerm = req.Term
+	votedFor = nil
+	response.Term = currentTerm
+
+	// TODO: Use pooler serving state to decide whether to vote
+	// Check if we're caught up with replication (within 30 seconds)
+	var lastMsgReceiptTime *time.Time
+	err := pm.db.QueryRowContext(ctx, "SELECT last_msg_receipt_time FROM pg_stat_wal_receiver").Scan(&lastMsgReceiptTime)
+	if err != nil {
+		// No WAL receiver (could be primary or disconnected standby)
+		// Don't reject the vote - let it proceed
+	} else if lastMsgReceiptTime != nil {
+		timeSinceLastMessage := time.Since(*lastMsgReceiptTime)
+		if timeSinceLastMessage > 30*time.Second {
+			// We're too far behind in replication, don't vote
+			return response, nil
+		}
+	}
+
+	// Grant vote and persist decision to LOCAL FILE (not Postgres!)
+	candidateID := &clustermetadatapb.ID{
+		Cell: pm.serviceID.GetCell(),
+		Name: req.CandidateId,
+	}
+
+	pm.mu.Lock()
+	pm.consensusTerm.VotedFor = candidateID
+	termToSave := pm.consensusTerm
+	pm.mu.Unlock()
+
+	if err := SetTerm(pm.config.PoolerDir, termToSave); err != nil {
+		return nil, fmt.Errorf("failed to persist vote: %w", err)
+	}
+
+	response.Accepted = true
+	return response, nil
+}
+
+// ConsensusStatus returns the current status of this node for consensus
+func (pm *MultiPoolerManager) ConsensusStatus(ctx context.Context, req *consensusdatapb.StatusRequest) (*consensusdatapb.StatusResponse, error) {
+	// Load consensus term from disk if not already loaded
+	pm.mu.Lock()
+	if pm.consensusTerm == nil {
+		term, err := GetTerm(pm.config.PoolerDir)
+		if err != nil {
+			pm.mu.Unlock()
+			return nil, fmt.Errorf("failed to load consensus state: %w", err)
+		}
+		pm.consensusTerm = term
+	}
+
+	// Get local voting term from file
+	localTerm := pm.consensusTerm.GetCurrentTerm()
+	pm.mu.Unlock()
+
+	// Get last successful leader term from Postgres (replicated data)
+	var leaderTerm int64
+	isHealthy := false
+
+	if pm.db != nil {
+		err := pm.db.QueryRowContext(ctx, `
+			SELECT COALESCE(MAX(leader_term), 0)
+			FROM multigres.heartbeat
+			WHERE shard_id = $1
+		`, []byte(req.ShardId)).Scan(&leaderTerm)
+
+		if err != nil {
+			pm.logger.Warn("Failed to query postgres leader term", "error", err)
+			leaderTerm = 0
+		} else {
+			isHealthy = true
+		}
+	}
+
+	// Get current WAL position
+	walPosition := &consensusdatapb.WALPosition{}
+	if isHealthy {
+		var err error
+		walPosition, err = pm.GetCurrentWALPosition(ctx)
+		if err != nil {
+			pm.logger.Warn("Failed to get WAL position", "error", err)
+		}
+	}
+
+	// Determine role (primary/replica)
+	role := "replica"
```

Comment:
Let' talk about this. It seems that we should be able to consolidate here with logic already implemented for PrimaryStatus / ReplicaStatus.